airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/token.py

```diff
@@ -22,92 +25,92 @@ class ApiKeyAuthenticator(AbstractHeaderAuthenticator):
     will result in the following header set on the HTTP request
     `"Authorization": "Bearer hello"`
 
+    Attributes:
+        header (Union[InterpolatedString, str]): Header key to set on the HTTP requests
+        refresh_token (Union[InterpolatedString, str]): Header value to set on the HTTP requests
+        config (Config): The user-provided configuration as specified by the source's spec
+        options (Mapping[str, Any]): Additional runtime parameters to be used for string interpolation
     """
 
-    def __init__(
-        self,
-        header: Union[InterpolatedString, str],
-        token: Union[InterpolatedString, str],
-        config: Config,
-        **options: Optional[Mapping[str, Any]],
-    ):
-        """
-        :param header: Header key to set on the HTTP requests
-        :param token: Header value to set on the HTTP requests
-        :param config: The user-provided configuration as specified by the source's spec
-        :param options: Additional runtime parameters to be used for string interpolation
-        """
-        self._header = InterpolatedString.create(header, options=options)
-        self._token = InterpolatedString.create(token, options=options)
-        self._config = config
+    header: Union[InterpolatedString, str]
+    refresh_token: Union[InterpolatedString, str]
```

Comment:
Language Change: In order to avoid a collision with the `token()` method that we need to implement, changing the incoming field name will work around that. The authenticator will now take in `header` and `refresh_token` fields.